### DISCUSSION TITLE
GLFW error #65542: GLX: GLX extension not found

### DISCUSSION THREAD

**Discussion by compressionist** [NVlabs/instant-ngp#344] [Q&A]
Hello everyone, 

I'm trying to run your GUI-enabled examples on a remote Ubuntu 20.04 virtual machine with a Quadro RTX 6000 (Lambda Labs entry-level instance). I installed the VNC server, VNC client and XQuartz on my Mac. So I can only open pictures in both tools and play videos in VNC. But unfortunately, when I run any of your examples with GUI support, I get the following error:

```
:~/instant-ngp$ ./testbed --scene data/image/tokyo.bin
05:24:09 INFO     Loading binary image from data/image/tokyo.bin
05:24:09 SUCCESS  Loaded a half-precision image with 1023x387 pixels.
05:24:09 INFO     Loading network config from: configs/image/base.json
05:24:09 INFO     GridEncoding:  Nmin=16 b=1.25984 F=2 T=2^24 L=16
05:24:09 INFO     Model:         2--[HashGrid]-->32--[FullyFusedMLP(neurons=64,layers=4)]-->3
05:24:09 INFO       total_encoding_params=1419392 total_network_params=7168
05:24:13 ERROR    GLFW error #65542: GLX: GLX extension not found
05:24:13 ERROR    Uncaught exception: GLFW window could not be created.
```

Maybe somebody of you solved similar issues?

The second question, if I run your examples without GUI, how do I understand that everything is working fine? How and where can I see intermediate and final results?

Best Regards,
Eugene

**Comment by Tom94** [comment]
Hi there,

I don't think GUI-over-VNC will be easy to make work. You might be able to get some remote OpenGL protocol working, but we don't officially support any of it.

While we're releasing this codebase as a way for other researchers to build upon and extend our results / tools, this is not fully fledged framework. There is no detailed documentation (exploring the source code is encouraged :)) and the scripts that we bundle only showcase basic functionality (especially without the GUI).

By running them with analogous command lines to the README, all you'll be able to see in console is that training doesn't explode (loss goes down), but not much more. `scripts/run.py` has a few convenience features, such as writing out screenshots or saving/loading snapshots of the trained weights, but nothing too in-depth.

In contrast, the Python bindings (of which you can see example use in `scripts/run.py` and a full list in `src/pythin_api.cu`) allow you to programmatically instrument *all* the features of __instant-ngp__. A good starting point would be to familiarize yourself with the different features of `scripts/run.py` (`python scripts/run.py --help`) and how they are implemented (screenshots, model snapshots, the training loop, etc.), followed by writing your own bespoke script that performs the experiments for you that you desire.

Cheers!

**Comment by compressionist** [comment]
@Tom94 Thank you so much, for the tips!

**Comment by hturki** [comment]
Hi @compressionist - one options that I've personally used to test things out over VNC is to install TurboVNC and VirtualGL. It's sometimes a bit finicky to set up but here are some rough instructions: https://gist.github.com/cyberang3l/422a77a47bdc15a0824d5cca47e64ba2

**Comment by compressionist** [comment]
Hi @hturki - thank you! I already did it, and it is working well.

**Reply by MatthewD1993** [reply]
Hi, could you tell how you solve it? I am facing the same problem, trying to use VNC to test this repo on a remote VM. Have you try TruboVNC, does it work? Thanks

**Reply by 1130xinxin** [reply]
您好，我在复现instant-ngp也遇到了同样问题，no gui情况下在哪能看到中间结果和最终结果？您是怎么解决这个问题的呢

**Comment by paulsung97** [comment]
I solved the GLX: GLX extension not found issue.
https://github.com/paulsung97/Solved-a-problem._GLFW_Error_-GLX-GLX_extension_not_found
